### PR TITLE
[Remove Stream and Optional usage:content-dashboard-document-library-impl]Lps 179646

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-179646

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by Kvale1733** [comment]
ci:test:echo

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java [hunk 1]
```diff
@@ -96,26 +95,33 @@ private String _getViewURL(Locale locale, ThemeDisplay themeDisplay) {
...
+
+			String backURL = ParamUtil.getString(
+				_httpServletRequest, "backURL");
+
+			if (Validator.isNotNull(backURL)) {
+				url = HttpComponentsUtil.setParameter(
+					friendlyURL, "p_l_back_url", backURL);
+			}
+
+			url = HttpComponentsUtil.setParameter(
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java:117]: @slnn This is wrong. The original logic just returns when `Validator.isNotNull(backURL)`, but your logic will make the value in the if-block never used.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java [hunk 2]
```diff
@@ -96,26 +95,33 @@ private String _getViewURL(Locale locale, ThemeDisplay themeDisplay) {
...
-			).orElse(
-				StringPool.BLANK
-			);
+					locale, clonedThemeDisplay);
+
+			if (friendlyURL == null) {
+				return StringPool.BLANK;
+			}
+
+			String url = null;
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java:107]: @slnn This variable is not needed. If you just keep re-assigning `friendlyURL` you will be able to simplify logic to
```
if (xxx) {
    return xxxxx1;
}

return xxxxx2;
```

Also, if friendlyURL is not null, the later operations will never return `null`, so the null check on line 120 is redundant.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/provider/FileExtensionGroupsProvider.java [hunk 3]
```diff
@@ -191,48 +190,39 @@ protected void activate(Map<String, Object> properties) throws Exception {
...
 
-		Stream<String> stream = Arrays.stream(mimeTypes);
+		List<String> extensions = new ArrayList<>();
 
-		String[] extensions = stream.flatMap(
-			mimeType -> {
-				Set<String> extensionsMimeTypes = _mimeTypes.getExtensions(
-					mimeType);
+		for (String mimeType : mimeTypes) {
+			extensions.addAll(
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/provider/FileExtensionGroupsProvider.java:209]: @slnn Please just use 2 layer for-loop.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/selector/ContentDashboardFileExtensionItemSelectorView.java [hunk 4]
```diff
@@ -133,28 +132,18 @@ private JSONArray _getContentDashboardFileExtensionGroupsJSONArray(
...
-				fileExtensionGroup -> fileExtensionGroup.toJSONObject(
-					SetUtil.fromArray(
-						servletRequest.getParameterValues(
-							"checkedFileExtensions")),
-					existingFileExtensions, _dlMimeTypeDisplayContext,
-					_language, themeDisplay.getLocale(), otherFileExtension)
-			).filter(
-				Objects::nonNull
-			).toArray());
+		Set<String> otherFileExtension = SetUtil.fromList(
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/selector/ContentDashboardFileExtensionItemSelectorView.java:135]: @slnn This variable name should be plural.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java [hunk 5]
```diff
@@ -386,26 +354,30 @@ public Date getModifiedDate() {
...
+		if (_group == null) {
+			return StringPool.BLANK;
+		}
 
-					return group.getName(locale);
-				}
+		String scopeName = null;
+
+		try {
+			scopeName = _group.getDescriptiveName(locale);
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java:364]: @slnn In the original logic, the try-catch is only meant to catch the exception declared by `group.getDescriptiveName`. And `group.getName()` will never return null.

So you logic can be simplified as

```
try {
    scopeName = xxxx;
}
catch (E e) {
    xxxxx;
}

if (scopeName == null) {
    return _group.getName(locale);
}

return scopeName;
```

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java [hunk 6]
```diff
@@ -585,36 +580,42 @@ private ContentDashboardItemVersion _getLastContentDashboardItemVersion(
...
+			getContentDashboardItemActions(
+				_portal.getHttpServletRequest(liferayPortletRequest),
+				ContentDashboardItemAction.Type.PREVIEW);
+
+		if (!contentDashboardItemActions.isEmpty()) {
+			ContentDashboardItemAction contentDashboardItemAction =
+				contentDashboardItemActions.get(0);
+
+			try {
+				url = new URL(contentDashboardItemAction.getURL());
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java:609]: @slnn Please simplify with direct returns.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java [hunk 7]
```diff
@@ -625,38 +626,53 @@ private String _getStringValue(String infoFieldName) {
...
+			serviceContext.getLiferayPortletRequest();
+
+		if (liferayPortletRequest == null) {
+			return null;
+		}
+
+		URL url = null;
+
+		try {
+			url = new URL(
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/FileEntryContentDashboardItem.java:663]: @slnn Also here.

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/provider/FileExtensionGroupsProviderTest.java [hunk 8]
```diff
@@ -60,18 +60,12 @@ public void testGetFileExtensionGroups() throws Exception {
...
 		List<FileExtensionGroupsProvider.FileExtensionGroup>
-			fileExtensionGroups =
-				fileExtensionGroupsProvider.getFileExtensionGroups();
-
-		Stream<FileExtensionGroupsProvider.FileExtensionGroup> stream =
-			fileExtensionGroups.stream();
-
-		Assert.assertTrue(
-			stream.filter(
+			filteredFileExtensionGroups = ListUtil.filter(
```
Comment (dantewang) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/test/java/com/liferay/content/dashboard/document/library/internal/item/provider/FileExtensionGroupsProviderTest.java:63]: @slnn You can directly use `ListUtil.exists`.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java (hunk 1 above)
Comment (slnn) [reply modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/ViewFileEntryContentDashboardItemAction.java:117]: oh, got it, my bad....

**Comment by slnn** [comment]
Resend in https://github.com/liferay-core-infra/liferay-portal/pull/4079

**Pull request closed without merging**